src/docket/exceptions.py

```diff
@@ -0,0 +1,2 @@
+class ForcedRetry(Exception):
+    """Raised when a task requests a retry via `in_` or `at`"""
```

Comment:
Maybe this should be in the same module next to the `Retry` dependency?  Having a module called `thing.exceptions` kind of implies there they're all in that module, but there are others that aren't here.  Also, not super convinced that all exceptions need to be grouped in a module unless there's a complex hierarchy to them